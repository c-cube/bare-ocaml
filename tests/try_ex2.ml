
module E = Example2
module Bare = Bare_encoding

let () =
  print_endline ">>> run encoding test 2";
  let p1 = E.PTreeNode {
      left=E.PTreeNil;
      person={E.Person.first="jean"; last="valjean"};
      right=E.PTreeNode{
          left=E.PTreeNil;
          person={E.Person.first="something something"; last="javert";};
          right=E.PTreeNil;
        };
    }
  in
  let s = Bare.to_string E.PTree.encode p1 in
  begin
    let oc = open_out "try2_foo.data" in
    output_string oc s; flush oc;
    close_out oc;
  end;
  let p2 = Bare.of_string_exn E.PTree.decode s in
  let s2 = Bare.to_string E.PTree.encode p2 in
  begin
    let oc = open_out "try2_foo2.data" in
    output_string oc s2; flush oc;
    close_out oc;
  end;
  assert (s = s2);
  assert (p1 = p2);
  ()

