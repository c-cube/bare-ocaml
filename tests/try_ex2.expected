>>> run encoding test 2
p1:
00000000: 0100 046a 6561 6e07 7661 6c6a 6561 6e01  ...jean.valjean.
00000001: 0013 736f 6d65 7468 696e 6720 736f 6d65  ..something some
00000002: 7468 696e 6706 6a61 7665 7274 00         thing.javert.

p2:
00000000: 0100 046a 6561 6e07 7661 6c6a 6561 6e01  ...jean.valjean.
00000001: 0013 736f 6d65 7468 696e 6720 736f 6d65  ..something some
00000002: 7468 696e 6706 6a61 7665 7274 00         thing.javert.

